doc/interoperability/pandas.rst

```diff
@@ -0,0 +1,299 @@
+>>> # last two switched order
+>>>
+>>> from sklearn.linear_model import LogisticRegression
+>>> lg = LogisticRegression(n_jobs=4, random_state=123, verbose=0,
+...                         penalty='l2', C=1.0,
+...                         solver='lbfgs', multi_class='auto')
+>>> lg.fit(train[feature_columns_train], train['target'])
+LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
+                   intercept_scaling=1, l1_ratio=None, max_iter=100,
+                   multi_class='auto', n_jobs=4, penalty='l2', random_state=123,
+                   solver='lbfgs', tol=0.0001, verbose=0, warm_start=False)
+>>>
+>>> prob1 = lg.predict_proba(test[feature_columns_train])
+>>> prob1[:5]
+array([[4.11775575e-03, 8.20651654e-01, 1.75230591e-01],
+       [9.42197387e-01, 5.78021174e-02, 4.96014488e-07],
+       [2.80792378e-07, 5.36112664e-03, 9.94638593e-01],
+       [6.92583081e-03, 7.79167290e-01, 2.13906879e-01],
+       [1.64170075e-03, 7.43560544e-01, 2.54797756e-01]])
+>>> # result is actually
+>>> prob2 = lg.predict_proba(test[feature_columns_test])
+>>> prob2[:5]
+array([[7.97326265e-01, 1.74068589e-01, 2.86051460e-02],
+       [9.96001477e-01, 3.99849230e-03, 3.08998201e-08],
+       [2.50514227e-03, 7.47234078e-03, 9.90022517e-01],
+       [7.15353463e-01, 2.33443544e-01, 5.12029929e-02],
+       [5.67215000e-01, 3.41931708e-01, 9.08532926e-02]])
+
+
+At the time of writing, it is the users responsibility to ensure that the
+column ordering in the data used for training the estimator is the same as the
+ordering of the data used for prediction. There is an ongoing discussion
+whether or not this will change in the future and this
+`issue <https://github.com/scikit-learn/scikit-learn/issues/7242>`__ should be
+watched and used to update this paragraph in the future. A simple and straight-
+forward way of ensuring that column ordering and column labels are the same is
+using something like :meth:`df.loc[:, list of column names]` to enforce the
+correct ordering.
+
+Handling Categorical data
+=========================
+
+Section to be extended.
+
+See the following references to get started:
+
+- https://scikit-learn.org/stable/glossary.html#term-categorical-feature
+- https://scikit-learn.org/stable/modules/preprocessing.html#preprocessing-categorical-features
+- https://github.com/scikit-learn-contrib/sklearn-pandas
+
+
+Dealing with heterogenous data
+==============================
+
+Many modern datasets used with Scikit-learn contain heterogenous data. For the
+purpose of adding bespoke preprocessing steps for separate columns, Scikit-
+learn provides an experimental :class:`~sklearn.compose.ColumnTransformer` API.
+This API (which might change in the future) allows the definition of different
+transformation steps to be applied to different columns in either arrays,
+sparse matrices or pandas DataFrames.
+
+Dealing with missing values
+===========================
```

Comment:
It might be good to mention here that the imputation methods now also support non-numeric data with the constant and most frequent strategies, and then link to the impute user guide.